src/main/java/rx/schedulers/ScheduleWhen.java

```diff
@@ -0,0 +1,263 @@
+ * actual {@link Scheduler}. When the {@link Worker} is unsubscribed the
+ * {@link Completable} emits an onComplete and triggers any behavior in the
+ * flattening operator. The {@link Observable} and all {@link Completable}s give
+ * to the flattening function never onError.
+ * <p>
+ * Slowing down the rate to no more than than 1 a second.
+ * 
+ * <pre>
+ * Scheduler slowSched = new ScheduleWhen(workers -> {
+ * 	// use concatenate to make each worker happen one at a time.
+ * 	return Completable.concat(workers.map(worker -> {
+ * 		// delay the starting of the next worker by 1 second.
+ * 		return worker.delay(1, TimeUnit.SECONDS);
+ * 	}));
+ * }, Schedulers.computation());
+ * </pre>
+ * <p>
+ * Limit the amount concurrency two at a time without creating a new fix size
+ * thread pool:
+ * 
+ * <pre>
+ * Scheduler limitSched = new ScheduleWhen(workers -> {
+ * 	// use merge max concurrent to two at a time
+ * 	return Completable.merge(workers, 2);
+ * }, Schedulers.computation());
+ * </pre>
+ */
+@Experimental
+public class ScheduleWhen extends Scheduler {
+	private final Scheduler actualScheduler;
+	private final PublishSubject<Observable<Completable>> workerQueue;
+
+	public ScheduleWhen(Func1<Observable<Observable<Completable>>, Completable> combine, Scheduler actualScheduler) {
+		this.actualScheduler = actualScheduler;
+		// workers are converted into completables and put in this queue.
+		this.workerQueue = PublishSubject.create();
+		// send it to a custom combinator to pick the order and rate at which
+		// workers are processed.
+		combine.call(workerQueue.onBackpressureBuffer()).subscribe();
+	}
+
+	@Override
+	public Worker createWorker() {
+		final Worker actualWorker = actualScheduler.createWorker();
+		// a queue for the actions submitted while worker is waiting to get to
+		// the subscribe to off the workerQueue.
+		final BufferUntilSubscriber<ScheduledAction> actionQueue = BufferUntilSubscriber.<ScheduledAction>create();
+		// convert the work of scheduling all the actions into a completable
+		Observable<Completable> actions = actionQueue.map(new Func1<ScheduledAction, Completable>() {
+			@Override
+			public Completable call(final ScheduledAction action) {
+				return Completable.create(new CompletableOnSubscribe() {
+					@Override
+					public void call(CompletableSubscriber actionCompletable) {
+						actionCompletable.onSubscribe(action);
+						action.call(actualWorker);
+						actionCompletable.onCompleted();
+					}
+				});
+			}
+		});
+
+		// a worker that queues the action to the actionQueue subject.
+		Worker worker = new Worker() {
+			private final AtomicBoolean unsubscribed = new AtomicBoolean();
+
+			@Override
+			public void unsubscribe() {
+				// complete the actionQueue when worker is unsubscribed to make
+				// room for the next worker in the workerQueue.
+				if (unsubscribed.compareAndSet(false, true)) {
+					actualWorker.unsubscribe();
+					synchronized (actionQueue) {
```

Comment:
Potential dead lock here as it will call user's Observer with a lock. You can just use SerializedSubject to wrap `actionQueue`.